### PR TITLE
feat(editor): bracket colorization

### PR DESCRIPTION
**Pull request by bbrabbitt**
I messed up with rebase in last PR #2604 so here is a new PR for bracket colorization. Anyway I had a few quite busy months that I don't have time to contribute to Lapce (I'm a PhD student) and I don't have time to work out a partial update mechanism for bracket colorization. Therefore I just add a limit so that files larger than the limit (default is 30000 line) will not be colorized. I have used it for some time and it works pretty well for me.

### CODE REVIEW

File: lapce-app/src/doc.rs [hunk 1]
```diff
@@ -238,6 +240,11 @@ impl Document {
             find_result: FindResult::new(cx),
             preedit: cx.create_rw_signal(None),
             common,
+            parser: Rc::new(RefCell::new(BracketParser::new(
+                "".to_string(),
```
Comment (panekj) [inline lapce-app/src/doc.rs:244]:
```suggestion
                String::new(),
```

File: lapce-app/src/doc.rs [hunk 2]
```diff
@@ -276,6 +284,11 @@ impl Document {
...
```
Comment (panekj) [inline lapce-app/src/doc.rs:288]:
```suggestion
                String::new(),
```

File: lapce-app/src/doc.rs [hunk 3]
```diff
@@ -315,6 +329,11 @@ impl Document {
...
```
Comment (panekj) [inline lapce-app/src/doc.rs:333]:
```suggestion
                String::new(),
```

File: lapce-app/src/doc.rs [hunk 4]
```diff
@@ -783,6 +809,25 @@ impl Backend for DocBackend {
         doc.retrieve_head();
     }
 
+    fn init_parser(doc: &Document<Self>) {
```
Comment (GreedyKomodoDragon) [inline lapce-app/src/doc.rs:812]: Just learning rust but could you do something like this?
```rs
let code = doc.buffer.get_untracked().to_string();

doc.backend.syntax.with_untracked(|syntax| {
    let syntax_styles = syntax.styles.is_some();

    let syntax_value = syntax_styles.then(|| syntax.clone());

    doc.backend.parser.borrow_mut().update_code(
        code,
        &doc.buffer.get_untracked(),
        syntax_value,
    );
});
```

Comment (panekj) [reply lapce-app/src/doc.rs:812]: This makes code harder to understand without inlay hints or knowing all libstd functions

**Comment by bbrabbitt** [comment]
Is the PR ready to merge or is there any more reviews?

**Comment by dzhou121** [comment]
Sorry for the delay. If you could make the option default to off, I can merge it.

**Comment by bbrabbitt** [comment]
> Sorry for the delay. If you could make the option default to off, I can merge it.

Sure, I turned it off.

**Pull request merged**